### DISCUSSION TITLE
[UI] Hedy rules in CodeMirror

### DISCUSSION THREAD

**Discussion by hasan-sh** [hedyorg/hedy#4890] [Large Projects & Plans]
**Idea incl level**
We want to apply indentation based on the commands the user types. Check https://github.com/hedyorg/hedy/pull/4832#pullrequestreview-1757762160 for more info. 

**Additional context**
I said "rules" just because this may not only be for indentation, but also other things. For instance, other code editors suggest commands when users click on tab and/or even autocomplete them; i.e., a for loop/function with a snippet of how it's used for a given level. Not sure if we need to, but for now, it's indentation only!

**Comment by Felienne** [comment]
I think this is a very interesting idea, but more of a discussion than an issue. Because there is potentially a tension between helping the user and teaching them things. I think this would be great for a BSc or maybe even MSc project where the student would explore how this feature would affect students, but not something we want to implement without a bit of exploration.
